neqo-http3/src/client_events.rs

```diff
@@ -181,11 +183,18 @@ impl SendStreamEvents for Http3ClientEvents {
 }
 
 impl ExtendedConnectEvents for Http3ClientEvents {
-    fn session_start(&self, connect_type: ExtendedConnectType, stream_id: StreamId, status: u16) {
+    fn session_start(
+        &self,
+        connect_type: ExtendedConnectType,
+        stream_id: StreamId,
+        status: u16,
+        headers: &Vec<Header>,
```

Comment:
Don't you want `&[Header]` for pass by reference?  (That would means `to_vec()` below)

However, given that you immediately and always clone, I think it would be better to have the caller decide whether a clone is needed, so this can take a value.

Reply:
Good point! I like this much better.